doc/source/administrator/authentication.md

```diff
@@ -0,0 +1,417 @@
+
+```yaml
+hub:
+  config:
+    Authenticator:
+      auto_login: true
+```
+
+### [enable_auth_state](https://jupyterhub.readthedocs.io/en/latest/api/auth.html#jupyterhub.auth.Authenticator.enable_auth_state)
+
+If you want JupyterHub to persist often sensitive information received as part
+of logging in, you need to enable it and provide one or more keys for encryption
+and decryption.
+
+The recommended way of doing so for this Helm chart is to configure
+[CryptKeeper](https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/crypto.py)
+with keys rather than setting an environment variable.
+
+For more information, see [JupyterHub's own
+documentation](https://jupyterhub.readthedocs.io/en/latest/reference/authenticators.html#authentication-state)
+about authentication state.
+
+```yaml
+hub:
+  config:
+    Authenticator:
+      enable_auth_state: true
+    CryptKeeper:
+      keys:
+        - 0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef
+```
+
+## Configuring authenticator classes
+
+Below we provide a few configuration examples of commonly used authentication
+classes. For more details about them, please see the authentication class' own
+documentation.
+
+### OAuth2 based authentication
+
+JupyterHub's [oauthenticator](https://github.com/jupyterhub/oauthenticator) has
+support for enabling your users to authenticate via a third-party OAuth2
+_identity provider_ such as GitHub, Google, and CILogon. All of these will
+require an OAuth2 _client id_ and _client secret_.
+
+For details on how to acquire a client id and client secret, please refer to
+[oauthenticator's
+documentation](https://oauthenticator.readthedocs.io/en/stable/getting-started.html).
+
+#### GitHub
+
+GitHub is the largest hosting service for git repositories. It is free to create
+an account at GitHub, and relatively straightforward to set up OAuth credentials
+so that users can authenticate with their GitHub username/password.
+
+To create OAuth credentials on GitHub, follow these steps:
+
+-   Click your profile picture -> settings -> developer settings
+-   Make sure you're on the "OAuth Apps" tab, then click "New OAuth App"
+-   Fill out the forms (you'll need your hub address) and generate your
+    ID/Secret.
+
+To enable GitHub authentication, your `config.yaml` should contain the following
+configuration:
+
+```yaml
+hub:
+  config:
+    OAuthenticator:
+      client_id: your-client-id
+      client_secret: your-client-secret
+      oauth_callback_url: https://your-jupyterhub-domain/hub/oauth_callback
+    JupyterHub:
+      authenticator_class: oauthenticator.github.GitHubOAuthenticator
+```
+
+Make sure that the `oauth_callback_url` matches the one you set in GitHub.
+
+To restrict access to the members of one or more GitHub organizations, amend
+your previous configuration with these parts.
+
+```yaml
+hub:
+  config:
+    GitHubOAuthenticator:
+      allowed_organizations:
+        - my-github-organization
+    OAuthenticator:
```

Comment:
While we *can* set every option on the highest possible class, it is not necessary. It's fine and probably easier to follow to only directly configure the class being used in many of these cases:

```suggestion
```